CMakeLists.txt

```diff
@@ -19,30 +19,51 @@ INCLUDE(CMakeDetectOsXArchs)
 PROJECT(s25client)
 
 IF (DEFINED CMAKE_TOOLCHAIN_FILE)
-	MESSAGE(STATUS "Using Toolchain definition file '${CMAKE_TOOLCHAIN_FILE}'")
+	MESSAGE(STATUS "Used Toolchain definition file '${CMAKE_TOOLCHAIN_FILE}'")
 ENDIF()
 
 STRING(TOLOWER ${CMAKE_SYSTEM_NAME} PLATFORM_NAME)
 SET(PLATFORM_NAME ${PLATFORM_NAME} CACHE STRING "Do not change" FORCE)
-SET(PLATFORM_ARCH ${CMAKE_SYSTEM_PROCESSOR} CACHE STRING "Do not change" FORCE)
+
+#Set PLATFORM_ARCH
+IF(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+	IF ("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+		SET(PLATFORM_ARCH "x86_64")
+	ELSE ("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+		SET(PLATFORM_ARCH "i386")
+	ENDIF ("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+ELSE()
+	IF(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i386" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i686")
+		SET(PLATFORM_ARCH "i386")
+	ELSEIF(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
+		SET(PLATFORM_ARCH "x86_64")
+	ELSEIF(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "universal" AND ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
```

Comment:
Was heißt "funktioniert nicht"? Build Server bauts und ich habe es gesetzt in der Platform datei...